pages/index.vue

```diff
@@ -78,19 +79,24 @@ export default Vue.extend({
           case 'latestmedia': {
             const latestMediaSections = [];
 
-            const userViewsRequest = await this.$api.userViews.getUserViews({
-              userId: this.$auth.user?.Id
-            });
+            let userViews: BaseItemDto[] = await this.getUserViews();
+
+            if (!userViews.length) {
+              await this.refreshUserViews();
+              userViews = await this.getUserViews();
+            } else {
+              this.refreshUserViews();
+            }
```

Comment:
```suggestion
            let refreshUserViewsPromise = this.refreshUserViews();
            if (!userViews.length) {
              await refreshUserViewsPromise;
              userViews = await this.getUserViews();
            }
```
I was bothered by the repetition, I guess this should solve it :p

Reply:
Updated to remove the repetition